src/Sentry/Span.cs

```diff
@@ -87,7 +95,15 @@ public Span(Transaction parentTransaction, SpanId? parentSpanId, string operatio
             _parentTransaction.StartChild(SpanId, operation);
 
         /// <inheritdoc />
-        public void Finish() => EndTimestamp = DateTimeOffset.UtcNow;
+        public void Finish()
+        {
+            EndTimestamp = DateTimeOffset.UtcNow;
+
+            if (Exception is {} exception)
+            {
+                _hub.BindException(exception, this);
```

Comment:
What if we have an overload to `Finish` that takes an Exception instead of the Exception property? Or am I missing a use case where just setting the exception without finishing is needed?

Reply:
We do have that overload, as an extension method.